test/tests/bootstrap/test_api20_os_bootstrap_parallel_proxy.py

```diff
@@ -0,0 +1,255 @@
+
+
+@attr(all=True)
+class api20_bootstrap_base(fit_common.unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # run all OS install workflows first
+        nodeindex = 0
+        for item in OSLIST:
+            # if OS proxy entry exists in RackHD config, run bootstrap against selected node
+            if proxy_select(item['path']) and nodeindex < len(NODECATALOG):
+                # delete active workflows for specified node
+                fit_common.cancel_active_workflows(NODECATALOG[nodeindex])
+                payload_data = {"options": {"defaults": {
+                                "version": item['version'],
+                                "kvm": item['kvm'],
+                                "repo": rackhdhost + proxy_select(item['path']),
+                                "rootPassword": "1234567",
+                                "hostname": "rackhdnode",
+                                "dnsServers": [rackhdconfig['apiServerAddress']],
+                                "users": [{"name": "rackhduser",
+                                           "password": "RackHDRocks!",
+                                           "uid": 1010}]}}}
+                # OS specific requirements
+                if item['workflow'] == "Graph.InstallUbuntu":
+                    payload_data["options"]["defaults"]["baseUrl"] = "install/netboot/ubuntu-installer/amd64"
+                    payload_data["options"]["defaults"]["kargs"] = {"live-installer/net-image": rackhdhost +
+                                                                    proxy_select(item['path']) +
+                                                                    "/ubuntu/install/filesystem.squashfs"}
+                # run workflow
+                result = fit_common.rackhdapi('/api/2.0/nodes/' +
+                                              NODECATALOG[nodeindex] +
+                                              '/workflows?name=' + item['workflow'],
+                                              action='post', payload=payload_data)
+                if result['status'] == 201:
+                    # this branch saves the task and node IDs
+                    NODE_STATUS[NODECATALOG[nodeindex]] = \
+                        {"workflow": item['workflow'],
+                         "version": item['version'],
+                         "kvm": item['kvm'],
+                         "id": result['json']['instanceId']}
+                    log.info_5(" TaskID: " + result['text'])
+                    log.info_5(" Payload: " + fit_common.json.dumps(payload_data))
+                else:
+                    # this is the failure case where there is no task ID
+                    NODE_STATUS[NODECATALOG[nodeindex]] = \
+                        {"workflow": item['workflow'],
+                         "version": item['version'],
+                         "kvm": item['kvm'],
+                         'id': "failed"}
```

Comment:
At this point there is no ID because the workflow failed to initialize. I could have put in another key pair but this was simpler, so this is intentional.